server/src/main/java/org/elasticsearch/common/geo/CentroidCalculator.java

```diff
@@ -31,29 +31,45 @@
 import org.elasticsearch.geometry.Point;
 import org.elasticsearch.geometry.Polygon;
 import org.elasticsearch.geometry.Rectangle;
+import org.elasticsearch.search.aggregations.metrics.CompensatedSum;
+
+import static org.elasticsearch.common.geo.DimensionalShapeType.LINE;
+import static org.elasticsearch.common.geo.DimensionalShapeType.POINT;
+import static org.elasticsearch.common.geo.DimensionalShapeType.POLYGON;
 
 /**
  * This class keeps a running Kahan-sum of coordinates
  * that are to be averaged in {@link TriangleTreeWriter} for use
  * as the centroid of a shape.
  */
 public class CentroidCalculator {
-    private double compX;
-    private double compY;
-    private double sumX;
-    private double sumY;
-    private double sumWeight;
+    CompensatedSum compSumX;
+    CompensatedSum compSumY;
+    CompensatedSum compSumWeight;
+
```

Comment:
Shouldn't we make these private as we have getters for them?

Reply:
I've removed the getters! thanks